src/Stratis.Bitcoin.IntegrationTests.Common/EnvironmentMockUpHelpers/NodeBuilder.cs

```diff
@@ -194,24 +126,23 @@ public CoreNode CloneStratisNode(CoreNode cloneNode)
         }
 
         /// <summary>A helper method to create a node instance with a non-standard set of features enabled. The node can be PoW or PoS, as long as the appropriate features are provided.</summary>
-        /// <param name="dataDir">The node's data directory where downloaded chain data gets stored.</param>
         /// <param name="callback">A callback accepting an instance of <see cref="IFullNodeBuilder"/> that constructs a node with a custom feature set.</param>
         /// <param name="network">The network the node will be running on.</param>
         /// <param name="protocolVersion">Use <see cref="ProtocolVersion.PROTOCOL_VERSION"/> for BTC PoW-like networks and <see cref="ProtocolVersion.ALT_PROTOCOL_VERSION"/> for Stratis PoS-like networks.</param>
-        /// <param name="configFileName">The name for the node's configuration file.</param>
         /// <param name="agent">A user agent string to distinguish different node versions from each other.</param>
-        public CoreNode CreateCustomNode(bool start, Action<IFullNodeBuilder> callback, Network network, ProtocolVersion protocolVersion = ProtocolVersion.PROTOCOL_VERSION, IEnumerable<string> args = null, string agent = "Custom")
+        /// <param name="configParameters">Use this to pass in any custom configuration parameters used to set up the CoreNode</param>
+        public CoreNode CreateCustomNode(bool start, Action<IFullNodeBuilder> callback, Network network, ProtocolVersion protocolVersion = ProtocolVersion.PROTOCOL_VERSION, string agent = "Custom", NodeConfigParameters configParameters = null)
         {
-            var argsList = args as List<string> ?? args?.ToList() ?? new List<string>();
+            configParameters = configParameters ?? new NodeConfigParameters();
 
-            string configFileName = "custom.conf";
-            if (!argsList.Any(a => a.StartsWith("-conf="))) argsList.Add($"-conf={configFileName}");
-            else configFileName = argsList.First(a => a.StartsWith("-conf=")).Replace("-conf=", "");
+            configParameters.SetDefaultValueIfUndefined("conf", "custom.conf");
+            string configFileName = configParameters["conf"];
 
-            string dataDir = this.GetNextDataFolderName(agent);
-            if (!argsList.Any(a => a.StartsWith("-datadir="))) argsList.Add($"-datadir={dataDir}");
+            configParameters.SetDefaultValueIfUndefined("datadir", this.GetNextDataFolderName(agent));
+            string dataDir = configParameters["datadir"];
 
-            return CreateNode(new CustomNodeRunner(dataDir, callback, network, protocolVersion, argsList, agent), start, configFileName);
+            configParameters.ToList().ForEach(p => this.ConfigParameters[p.Key] = p.Value);
```

Comment:
This can go in `NodeConfigParameters` as a method.

Reply:
OK I'll do that

Reply:
So in line 145 you can do
`return CreateNode(new CustomNodeRunner(dataDir, callback, network, protocolVersion, configParameters, agent), start, configParameters["conf"]);`

Reply:
OK I see... honestly yes I can, I can also not declare dataDir, I am just confused : what point are you trying to make ?

Reply:
oh it's just a suggestion, nothing more.